### ISSUE TITLE
Add "Allocation" feature for better link validation

### ISSUE THREAD

**Issue by robertberrington** [jacebrowning/doorstop#194]
An `item.allocation` could be added to allow the user to list the Child documents where requirements should link.

A `WARNING` can be thrown if:
  * The Parent document is missing a Child item linking to it
  * The Child document link points to a Parent whose allocation does not include the child

This would allow a more intelligent `WARNING` for `no links from child document` to only list the documents where links should exist.

It seems a simple "space-delimited" field would be sufficient.

Consider an `ERROR` if the document ID does not exist.

**Comment by jacebrowning** [comment]
This feature request is mostly clear, but I have a few questions.

> An `item.allocation` could be added to allow the user to list the Child documents where requirements should link.

Is this attribute now required on all all items?

> * The Parent document is missing a Child item linking to it

Do you mean "Parent **item**"?

> It seems a simple "space-delimited" field would be sufficient.

I'd probably go with a YAML list for simplicity of parsing. What if this attribute is left blank? Are zero links expected?

**Comment by robertberrington** [comment]
@jacebrowning 

> Is this attribute now required on all all items?

Yes, but only `normative=TRUE` should be populated. Perhaps throw a fault if `normative=TRUE` and item.allocation is empty. Also, throw a fault if `normative=FALSE` and item.allocation is NOT empty.

I was thinking really hard if I could come up with an instance where a requirement stops someplace other than at the bottom of the children. That would mean that someone is implementing a design/test on an item that isn't supported by underlying children requirements/implementation/test. I think "flow-down" requirements exist because the child needs to link to the parent, but there's no change.

> Do you mean "Parent item"?

Yes.

> I'd probably go with a YAML list for simplicity of parsing. What if this attribute is left blank? Are zero links expected?

Sure ... I was thinking that the export to Excel would produce a space-delimited list, and that on import, the space-delimited list would be parsed.

I suppose another thing to consider is if a user decides NOT to use this feature ... do you want to give them option for a `--no-alloc-check` ... in which case you'd have the current behavior?

**Comment by jacebrowning** [comment]
> Do you want to give them option for a `--no-alloc-check`?

Yeah, that would be good.
